packages/frontend/src/util/errors.tsx

```diff
@@ -0,0 +1,36 @@
+import Dialog, { Content, Portal } from "@corvu/dialog";
+
+import "./errors.css";
+
+export class PermissionsError extends Error {
+    constructor(message: string) {
+        super(message);
+        this.name = "AuthorizationError";
+    }
+}
+
+export function ErrorBoundaryDialog(props: { error: Error }) {
+    console.error(props.error);
+
+    let heading: string;
+    let message: string;
+
+    if (props.error instanceof PermissionsError) {
+        heading = "Permissions Error";
+        message = "You are not permitted to view this resource.";
+    } else {
+        heading = "Error";
+        message = "An unknown error occurred.";
```

Comment:
I figure that once the known error types (currently just permissions error, maybe in the future additional types) have been exhausted, the error presumably arises from a bug in our application rather than any expected behavior. In this case, perhaps we should warn the user that something bad has happened and tell them how to file a bug report (file an issue on GitHub). That come could in a later PR, though.

Reply:
I created an issue for instructing users to file bug reports [here](https://github.com/ToposInstitute/CatColab/issues/403)

On this topic, it can be beneficial to log client side errors on the backend for a number of reasons, I created a ticket for that [here](https://github.com/ToposInstitute/CatColab/issues/404).